### DISCUSSION TITLE
Beta build notes - Mike KQ9P

### DISCUSSION THREAD

**Discussion by MikeMcPherson** [alanbjohnston/CubeSatSim#326] [General]
I'll be posting my notes from building the beta boards that Alan so generously sent out.  I hope my feedback isn't too late to be helpful.

**Comment by MikeMcPherson** [comment]
STEM Payload Board done.  As expected, the instructions are excellent and the build was mostly painless.  The board is running, sending beautiful transmissions, and FoxTelem is having no trouble decoding the telemetry.  (Pardon my flux; I haven't cleaned anything yet.)

![IMG_2463](https://github.com/user-attachments/assets/682d7c53-4502-4070-806f-dfed18efde04)
![IMG_2461](https://github.com/user-attachments/assets/03386e61-ad0f-499e-8ec8-48135e6813b8)

**Comment by Boovtech** [comment]
Not too late at all.  I have not even started building.  I do have a loaner cubesat sim here that I am trying to troubleshoot.  It has some power issues.  We get one one here in DFW  ocassion to demonstrate at a couple of summer events and I have been given permission to take a look at it.  Working on one that is already built is helpful in itself.  It is quite different from the previous one I built.  

I'm looking forward to your notes.

Virginia NV5F

**Comment by MikeMcPherson** [comment]
The biggest issue I ran into is the spurious traces on the two SMA connector footprints, the one on X3 worse than the one on X2:

![IMG_2456](https://github.com/user-attachments/assets/bcb0a1a2-3cfa-4f33-8b09-9e2dd3b00cfc)

As expected, on X3 the connector housing shorted to the spurious trace and hence to the center pin.  I removed the trace and everything is fine:

![IMG_2459](https://github.com/user-attachments/assets/db058022-1370-4f93-9508-9fe919a3f543)

There are other similar short bits of trace that aren't in problematic locations.  Are these registration marks or some such thing?

**Reply by alanbjohnston** [reply]
This SMT pad on the top of the PCB is for mounting the optional E1 and E2, the 433 MHz chip antennas which goes from there to the edge of the PCB.  They can be used in place of the SMA connectors or the tape measure antenna that screws directly onto the center of the SMA pad.

In your first photo, it isn't bridged, and the SMA connector is soldered on the PCB bottom, so there won't be any accidental bridging with it when you solder the SMA connector.

PCBWay does put some solder paste on it, which isn't necessary.  I can have them not put any solder paste on it going forward.

The other short bits of trace you mention are probably unused SMT pads.  Most parts on the PCB have a through hole and and surface mount pad so it can be built either way.

**Reply by MikeMcPherson** [reply]
I see what you're saying, but the body of the SMA connector rested on the E1 trace and shorted the connector.  I expect if the trace had been centered on the hole for the SMA center conductor - as it in the wiki pictures - there would have been no problem.  So I guess my comment changes to: the E1 trace that crosses the SMA center conductor hole is offset on my board and shorts to the body of the SMA connector.

**Comment by MikeMcPherson** [comment]
A documentation suggestion: I can't tell whether the Pico simply replaces the Blue Pill and the Pi Zero is still required, or whether the Pico can replace both the Pi Zero and the Blue Pill.

**Reply by Boovtech** [reply]
In this version the PI Zero is still required and the Pi pico is now the microcontroller for the payload board instead of the Blue Pill or the Sparkfun Aurduino.

**Reply by alanbjohnston** [reply]
Virginia is correct that the Raspberry Pi Zero is needed - it runs the housekeeping telemetry software and controls the transmitter and the command and control receiver.  The Raspberry Pi Pico microcontroller replaces the Sparkfun Pro Micro or STM32 "blue pill" in the original v1.2 design.  The Pico only talks to the BME and MPU sensors, talks to a serial GPS module if it is wired up to RX2, and also any additional sensors that you add.  The Raspberry Pi Pico software can be modified using the Arduino IDE.

**Comment by MikeMcPherson** [comment]
More documentation: on one computer I can right-click images in the build instructions, select "Open Image in New Tab", and I see the full-resolution version of the image.  On another computer the new tab is black, no image.  Both computers are running Windows 10 with the latest patches, both running Firefox.  In both cases "Save Image As..." works fine.  Once I get the boards built I'm going to dig deeper and see if I can figure out what's different between the two computers.

**Reply by alanbjohnston** [reply]
This is a Github issue - their image hosting for the wiki is all messed up - if you Google it you will find lots of complaints.  You might have also noticed that sometimes random images on a wiki page will fail to load.  It is their buggy software, not a particular browser or internet connection.  It will be fine one day, and bad the next. Hopefully Github will eventually fix their bugs.

The only solution I have come up with is to post a PDF of each wiki page - there is a link at the top of each page.  These PDFs have all the images and you should be able to zoom in on them.  You can see them where it says "If the images on this page fail to load, you can download a PDF of this page here."

**Reply by MikeMcPherson** [reply]
Got it, blew right past that first line...

**Comment by MikeMcPherson** [comment]
Finally, when I fired up the STEM Payload Board with just the Pi Zero W in Step 2, I heard a high-pitched whining sound when the board was transmitting.  According to the spectrum analyzer app on my iPhone, it was at about 4935 Hz.  Now that the board is fully-populated and finished I don't hear it any more.

**Reply by alanbjohnston** [reply]
Most sounds like that are due to switching power supplies, and the Pi Zero has one built-in which generates the 3.3V and 1.8V needed for the CPU.  The particular load at that time might have produced the audible artifact. The previous design of the CubeSatSim had an extra boost converter module in it, which sometimes made noise.  The current CubeSatSim design doesn't add any switching power supply in the battery and charging circuit, so it shouldn't make noise.  Note that the Raspberry Pi Pico also has a built in switching power supply too, so it may make sounds under certain conditions.

Thanks for all this feedback! :-D

Alan

**Comment by MikeMcPherson** [comment]
Out of habit - and experience with bad INA breakouts on earlier builds - I started to solder female headers for the INA219 breakouts.  I see, though, that you aren't recommending that any more.  What's your thinking on shifting to soldering direct to the board?

**Reply by alanbjohnston** [reply]
I haven't had a failure of the blue INA219 boards in a long time.  I have had some intermittent issues that turned out to be the female headers.  So I don't recommend using sockets anymore or pre-testing the blue INA219 boards anymore.  I recommend soldering the blue INA219 boards directly to the PCB.

**Reply by MikeMcPherson** [reply]
Sounds good.  Thanks.

**Comment by MikeMcPherson** [comment]
Checklist Step 6: Solar Board.  The second line says "D1-D7", should be "D1-D6".

**Reply by alanbjohnston** [reply]
Fixed!

**Comment by MikeMcPherson** [comment]
Step 5: Battery Board.  It's noted in the BOM that you can substitute two BT1 if BT2 is out of stock (which it was when I ordered).  In the Step 5 instructions you should mention that if you have three BT1 holders you'll need to trim the tabs off the bottom of two of them and be careful to get those two in the correct orientation.

**Comment by MikeMcPherson** [comment]
![IMG_2466](https://github.com/user-attachments/assets/edba55a5-ca84-4d18-afc6-ad1398c0f1a0)

**Reply by MikeMcPherson** [reply]
The build was uneventful.  Superficially, it is working fine.  I'm getting telemetry and switching between modes works.  I need to set this aside for a bit and work on something else.  I'll do more thorough testing later.
